plugins/@grouparoo/demo/src/mongo/connection.ts

```
@@ -90,6 +93,47 @@ function findConfig() {
   return { uri, database };
 }
 
+function castRow(
+  collectionName,
+  row
+): { [key: string]: string | number | Date } {
+  const keys = Object.keys(row);
+  for (const key of keys) {
+    row[key] = castValue(collectionName, key, row[key]);
+  }
+  return row;
+}
+
+function castValue(
+  collectionName,
+  field,
+  value
+): string | number | boolean | Date {
+  if (value === null || value === undefined) {
+    return null;
+  }
+  const type = TYPES[collectionName]?.validator?.$jsonSchema?.properties[field];
+  if (!type) {
+    return null;
```

maybe `throw` here because it's unexpected.